CMakeLists.txt

```diff
@@ -43,16 +43,159 @@ target_include_directories(adolc
+# independent variables will be recorded on the trace, this however
+# requires more checks to be performed during the tracing and increases
+# tracing time. Useful only if memory is a constraint and tracing is
+# done fewer times than derivative computations.
+option(ENABLE_ACTIVITY_TRACKING "Enable activity tracking to reduce trace size but increased tracing time" FALSE)
+add_feature_info(ENABLE_ACTIVITY_TRACKING ENABLE_ACTIVITY_TRACKING "Activity tracking to reduce trace size but increased tracing time.")
+target_compile_definitions(adolc PRIVATE
+  $<$<BOOL:${ENABLE_ACTIVITY_TRACKING}>:ADOLC_TRACK_ACTIVITY=1>)
+
+# Whether to use ADOL-C hard debug mode
+option(ENABLE_HARDDEBUG "Enable ADOL-C hard debug mode" FALSE)
+add_feature_info(ENABLE_HARDDEBUG ENABLE_HARDDEBUG "Hard debug mode.")
+if (ENABLE_HARDDEBUG)
+  set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "Choose the type of build." FORCE)
+endif (ENABLE_HARDDEBUG)
+target_compile_definitions(adolc PRIVATE
+  $<$<OR:$<CONFIG:Debug>,$<BOOL:${ENABLE_HARDDEBUG}>>:ADOLC_DEBUG=1>
+  $<$<BOOL:${ENABLE_HARDDEBUG}>:ADOLC_HARDDEBUG=1>)
+
+# The adouble default constructor does not initialize the value to zero
+# (improves performance but yields incorrect results for implicit array initializations, see manual)
+option(DISABLE_STDCZERO "adouble default constructor does not initialize the value to zero" FALSE)
+add_feature_info(DISABLE_STDCZERO DISABLE_STDCZERO "Do not initialize the values to zero.")
+target_compile_definitions(adolc PRIVATE
+  $<$<NOT:$<BOOL:${DISABLE_STDCZERO}>>:ADOLC_ADOUBLE_STDCZERO=1>)
 
-set(ADVBRANCH "#undef ADOLC_ADVANCED_BRANCHING")
-set(ADTL_REFCNT "#undef USE_ADTL_REFCOUNTING")
+# adouble constructors need to be called. With malloc or realloc that is not the case.
+# With this option the adouble can do a late initialization.)
+option(ENABLE_LATEINIT "Whether the adouble variables have a late initialize option for functions like malloc/realloc" FALSE)
+add_feature_info(ENABLE_LATEINIT ENABLE_LATEINIT "adouble can do a late initialization.")
+target_compile_definitions(adolc PRIVATE
+  $<$<BOOL:${ENABLE_LATEINIT}>:ADOLC_ADOUBLE_LATEINIT=1>)
+
+# Whether errno is thread save
+option(ENABLE_TSERRNO "Use errno as thread number cache" FALSE)
+add_feature_info(ENABLE_TSERRNO ENABLE_TSERRNO "Use errno as thread number cache.")
+target_compile_definitions(adolc PRIVATE
+  $<$<BOOL:${ENABLE_TSERRNO}>:ADOLC_THREADSAVE_ERRNO=1>)
+
+# Whether to enable OpenMP
+option(ENABLE_OPENMP "Enable OpenMP" FALSE)
+add_feature_info(ENABLE_OPENMP ENABLE_OPENMP "Enable OpenMP.")
+if(ENABLE_OPENMP)
+  find_package(OpenMP REQUIRED)
+  target_link_libraries(adolc PUBLIC OpenMP::OpenMP_CXX)
+endif()
+
+
+find_package(Boost 1.54 COMPONENTS system)
+set(WITH_BOOST 0)
+if(Boost_FOUND)
+  target_link_libraries(adolc PUBLIC Boost::boost Boost::system)
+  set(WITH_BOOST 1)
+  # TODO: check for header boost/pool/pool_alloc.hpp
+  set(USE_BOOST_POOL "#define USE_BOOST_POOL 1")
```

Comment:
If I got this correctly, boost-pool is used whenever Boost system was found, right? We might have a user-defined option whether to use boost-pool or not.